olive/data_config/config.py

```diff
@@ -0,0 +1,193 @@
+                self.components[k].name = self.components[k].name or v.name
+                self.components[k].params = self.components[k].params or v.params
+
+    def _update_default_component_type(self):
+        """
+        Resolve the default component type.
+        """
+        dc_cls = Registry.get_container(self.type)
+        self.default_components_type = dc_cls.default_components_type or {}
+        # 1. update default_components_type with task_type for huggingface case
+        self._update_default_component_type_with_task_type(dc_cls)
+        # 2. update default_components_type with DefaultDataComponentCombos
+        # for those components not defined in the container config
+        for k, v in DefaultDataComponentCombos.items():
+            if k not in self.default_components_type:
+                self.default_components_type[k] = v
+
+    def _update_default_component(self):
+        """
+        Resolve the default component type.
+        """
+        for k, v in self.default_components_type.items():
+            self.default_components[k] = DataComponentConfig(type=v, name=v, params={})
+
+    def fill_in_params(self):
+        """
+        Fill in the default parameters for each component.
+        1. if prams_config is not None, use the params_config to fill in the params
+        2. if params_config is None, use the default params from the function signature
+        3. if there is already define params under the component, use the params directly
+        """
+        from inspect import signature
+
+        self.params_config = self.params_config or {}
+        for k, v in self.components.items():
+            component = Registry.get_component(k, v.type)
+            # 1. user function signature to fill params firstly
+            params = signature(component).parameters
+            for param, info in params.items():
+                # 2. override the params with params_config
+                if param in self.params_config:
+                    v.params[param] = self.params_config[param]
+                    continue
+                # 3. if it already defined params under the component, use the params directly
+                if param not in v.params and not param.startswith("_"):
+                    if info.kind == info.VAR_POSITIONAL or info.kind == info.VAR_KEYWORD:
+                        continue
+                    elif info.default is info.empty:
+                        logger.debug(f"Missing parameter {param} for component {k}")
+                        v.params[param] = None
+                    else:
+                        v.params[param] = params[param].default
+
+    def get_components_params(self):
+        """
+        Get the parameters of the data container.
+        """
+        return {k: v.params for k, v in self.components.items()}
+
+    @property
+    def dataset(self):
+        """
+        Get the dataset of the data container.
```

Comment:
>Get the dataset of the data container. [](http://example.com/codeflow?start=8&length=38)

Get the torch dataset from Olive dataset

Reply:
Updated.